aws/resource_aws_glue_data_catalog_encryption_settings.go

```diff
@@ -0,0 +1,188 @@
+package aws
+
+import (
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/glue"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func resourceAwsGlueDataCatalogEncryptionSettings() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsGlueDataCatalogEncryptionSettingsPut,
+		Read:   resourceAwsGlueDataCatalogEncryptionSettingsRead,
+		Update: resourceAwsGlueDataCatalogEncryptionSettingsPut,
+		Delete: schema.Noop,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"catalog_id": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Optional: true,
+				Computed: true,
+			},
+			"data_catalog_encryption_settings": {
+				Type:     schema.TypeList,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"connection_password_encryption": {
+							Type:     schema.TypeList,
+							Required: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"aws_kms_id": {
```

Comment:
Change to `aws_kms_key_id` to match API.

Reply:
fixed